src/string_utils.h

```diff
@@ -104,4 +104,18 @@ static bool try_stoull(unsigned long long& val, const std::string& str, std::siz
     return false;
 }
 
+static float parse_float(const std::string& s, std::size_t* float_len = nullptr){
+    std::stringstream ss(s);
+    ss.imbue(std::locale::classic());
+    float ret;
+    ss >> ret;
+    if(ss.fail()) throw std::invalid_argument("parse_float: Not a float");
+    if(float_len != nullptr){
+        auto pos = ss.tellg();
+        if(ss.fail()) *float_len = s.size();
```

Comment:
This fail() doesn't get checked if it throws before it.

Reply:
The second fail check refers to `tellg`. If the `stringstream` is at the end of the string and you call `tellg` on it, it is supposed to return -1. It will also set the failbit.